src/forwardproblem.cpp

```diff
@@ -113,6 +125,15 @@ void ForwardProblem::workForwardProblem() {
         }
     }
 
+    if(edata && edata->fixedParameters.size()) {
+        // fixed parameter in model are superseded by those provided in edata
+        // Note: this changes those parameters of `model` permanently
+        if(edata->fixedParameters.size() != (unsigned) model->nk())
+            throw AmiException("Number of fixed parameters (%d) in model does not match ExpData (%zd).",
+                               model->nk(), edata->fixedParameters.size());
+        model->setFixedParameters(edata->fixedParameters);
+    }
```

Comment:
I think if `edata->fixedParameters.size()` is not provided here, we probably want to revert back to the model specified parameters that we may have overwritten with `edata->fixedParameters`.

Reply:
You are right.

Reply:
(Overwritten by `edata->fixedParametersPreequilibration` you mean?)

Reply:
yes